### ISSUE TITLE
[Performance] Make the Search Sidebar transition relatively smooth on slower devices 

### ISSUE THREAD

**Issue by mountiny** [Expensify/App#33596]
### Problem

coming from [here](https://expensify.slack.com/archives/C05LX9D6E07/p1699514971219659)

When user navigates back from Search Sidebar there’s a noticeable delay and lag in closing the Search Sidebar. It mainly arises from withNavigationFocus HOC as it re-renders the BaseOptionSelector tree including OptionsList . Adding a memo to OptionsList doesn’t help as it’s parent gets re-rendered due to the withNavigationFocus HOC.

Also, we notice that due to isFocused state value in ActionItemView the ReportActionsList re-renders which also adds the overhead to the rendering cycle as it recreates the instances of recordTimeToMeasureItemLayout and loadOlderChats which are passed as props to ReportActionsList .

### Solution

We can remove the withNavigationFocus HOC and add navigation listeners(focus, blur) in BaseOptionSelector and memoize OptionsListto not re-render when state is set in it’s parent. We can avoid recreating instances of recordTimeToMeasureItemLayout and loadOlderChats by wrapping them in useCallbacks and memoizing ReportActionsList .

**Comment by mountiny** [comment]
@hurali97 Could you comment on this one?

**Comment by mountiny** [comment]
Pr is close to be merged

**Comment by hurali97** [comment]
Hey @mountiny 👋

**Comment by hurali97** [comment]
> @hurali97 Could you comment on this one?

Is there something else should I comment ? or just the above is fine :D

**Comment by mountiny** [comment]
Haha all good i just cannot assign external people who did not comment on the issue ;)

**Comment by mountiny** [comment]
PR was reverted because there was couple of regressions with lists, we need to address them in a new PR

**Comment by hurali97** [comment]
@mountiny Following are the regression tickets that needs addressing with new PR for the Search Improvement, correct?

- https://github.com/Expensify/App/issues/34241
- https://github.com/Expensify/App/issues/34252
- https://github.com/Expensify/App/issues/34239 ( PR created )

**Comment by situchan** [comment]
> @mountiny Following are the regression tickets that needs addressing with new PR for the Search Improvement, correct?
> 
> * [Expense - "To" field appears with delay in confirmation page #34241](https://github.com/Expensify/App/issues/34241)
> * [ IOU - Category page renders slowly #34252](https://github.com/Expensify/App/issues/34252)
> * [[$125] Request money - The 'Enter' key fails to select the email/workspace/phone when requesting money. #34239](https://github.com/Expensify/App/issues/34239) ( PR created )

@hurali97 here's candidates: (some are not confirmed but possibly caused by our PR)
- https://github.com/Expensify/App/issues/34239
- https://github.com/Expensify/App/issues/34241
- https://github.com/Expensify/App/issues/34249
- https://github.com/Expensify/App/issues/34252
- https://github.com/Expensify/App/issues/34269
- https://github.com/Expensify/App/issues/34271
- https://github.com/Expensify/App/issues/34285

(most of them have the same root cause)

**Comment by hurali97** [comment]
@situchan  Thanks, I will see them 👍

**Comment by bfitzexpensify** [comment]
I'm going to be mostly ooo until Jan 29, assigning a second BZ team member to keep an eye on this.

- ready for payment on 01/18
- BZ checklist needs completing (bump on that @situchan)

**Comment by adelekennedy** [comment]
created a manual contract for @situchan

**Comment by situchan** [comment]
This is not ready for payment yet. 2nd PR is in review

**Comment by adelekennedy** [comment]
huh - updating the title

**Comment by situchan** [comment]
Please remove HOLD for payment from issue title

**Comment by mountiny** [comment]
Not overdue, pr is at works

**Comment by mountiny** [comment]
PR is at works

**Comment by situchan** [comment]
PR is in review

**Comment by situchan** [comment]
still in review

**Comment by bfitzexpensify** [comment]
Looks like the PR has hit prod - https://github.com/Expensify/App/pull/34422. 

I'm not sure the automation will work here, so updating the title manually. 

As for payment - given the original PR was reverted due to regressions, it seems like the payout here would be $250. 

Agree with that @mountiny?

**Comment by mountiny** [comment]
I would say that this was a complex change to test and the regressions were very hard to catch, I would still payout $500 to @situchan as this was challenging task taking lots of time. @hurali97 is from agency.

**Comment by bfitzexpensify** [comment]
Cool, sounds good to me. Offer sent @situchan

**Comment by bfitzexpensify** [comment]
Payment due Friday

**Comment by bfitzexpensify** [comment]
Payment complete - @situchan can you please complete the BZ checklist? Will copy it into a new comment

**Comment by bfitzexpensify** [comment]
BugZero Checklist: The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ]  [@situchan] The PR that introduced the bug has been identified. Link to the PR:
- [ ]  [@situchan] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ]  [@situchan] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ]  [@situchan] Determine if we should create a regression test for this bug.
- [ ]  [@situchan] If we decide to create a [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) for the bug, please propose the regression test steps to ensure the same bug will not reach production again.
- [ ]  [@bfitzexpensify ] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by situchan** [comment]
This is performance improvement, not bug. BZ Checklist doesn't apply here.

**Comment by muttmuure** [comment]
I think this can be closed
